gazebo_ros/launch/gzclient.launch.py

```diff
@@ -54,23 +69,28 @@ def generate_launch_description():
                               description='Set "true" to run gzserver with gdb'),
         DeclareLaunchArgument('valgrind', default_value='false',
                               description='Set "true" to run gzserver with valgrind'),
+        DeclareLaunchArgument('gui_required', default_value='false',
+                              description='Set "true" to shut down launch script when GUI is terminated'),
 
-        # Execute
+        # Execute node with Shutdown if required
```

Comment:
It really helped me here for undestanding the on_exit + condition combo your explanation in the pull request description:
```

Better alternative

The code is kind of ugly right now. It duplicates ExecuteProcess(). One call has IfCondition(LaunchCOnfiguration('server_required')) and on_exit=Shutdown() in the case the node is required, e.g. server_required:=true. The other call has UnlessCondition in the case the node is not required. That is based on this ROS Answers example. It's similar to the <group if> in ROS 1 XML launch.

I've asked William whether there's a better way now, and he doesn't think we've implemented a better way. Probably this has to be implemented ros2/launch#290 for a cleaner way to exist. #962 also works around it.

The ROS 2 XML launch does have <group if>. But since the launch script is already in Python and XML launch doesn't exist before Eloquent, I just stuck to the Python script.
```
Can we add a pointer here in the code or a short version of it?

Reply:
I added a bit of description and link in b393d5f